ParameterTrendPercentages/ParameterTrendPercentages_1/ParameterTrendPercentages_1.cs

```diff
@@ -209,31 +208,92 @@ private void RetrieveTrendRecords()
         };
 
         var getTrendDataResponseMessage = (GetTrendDataResponseMessage)_callback.SendMessage(getTrendDataMessage);
-        List<TrendRecord> records = getTrendDataResponseMessage.Records.Values.FirstOrDefault().OrderBy(r => r.Time).ToList() ?? throw new DataMinerException("Failed to retrieve trend information for the provided timeslot");
+        var response = getTrendDataResponseMessage;
+
+        // Process trending information
+        Dictionary<string, double> result = new Dictionary<string, double>();
+        if (response.Records.Values.FirstOrDefault() != null)
+        {
+            List<TrendRecord> records = response.Records.Values.FirstOrDefault().OrderBy(r => r.Time).ToList();
+            Dictionary<DateTime, string> trendRecords = new Dictionary<DateTime, string>();
+
+            // Search all records before start time and add closest with updated start time
+            List<TrendRecord> beforeStartRecords = records.Where(r => r.Time <= _start).ToList();
+            trendRecords.Add(_start, beforeStartRecords.Count > 0 ? beforeStartRecords.Last().GetStringValue() : "Not trended");
 
-        // Remove empty values or outside window of request
-        List<TrendRecord> filteredRecords = records
-            .Where(r => r.Time >= _start && r.Time <= _end && !string.IsNullOrEmpty(r.GetStringValue()))
-            .ToList();
+            // Add trend records to Dictionary if value is not empty and within window
+            foreach (var record in records.Where(r => r.GetStringValue() != string.Empty && r.Time > _start && r.Time < _end))
```

Comment:
All records before the start are still in here while you have already pinpointed the index at which you overlap with the time range.